pkg/plugin/aggregation/aggregator_test.go

```diff
@@ -286,56 +289,58 @@ func TestAggregation_RetryWindow(t *testing.T) {
-		if tc.simulateErr {
-			r = iotest.TimeoutReader(testDataReader)
-		} else {
-			r = strings.NewReader("foo")
-		}
+		t.Run(tc.desc, func(t *testing.T) {
+			agg := NewAggregator(dir, expected)
+			// Shorten retry window for testing.
+			agg.retryWindow = testRetryWindow
+			testDataPath := "./testdata/fakeLogData.txt"
+			testDataReader, err := os.Open(testDataPath)
+			if err != nil {
+				t.Fatalf("Could not open test data file: %v", err)
+			}
+			defer testDataReader.Close()
 
-		err = agg.processResult(&plugin.Result{Body: r, NodeName: "node1", ResultType: "systemd_logs"})
-		if err == nil && tc.simulateErr {
-			t.Fatal("Expected error processing this due to reading error, instead got nil.")
-		}
-		// check time before/after wait and ensure it is greater than the retryWindow.
-		time.Sleep(tc.postProcessSleep)
-		start := time.Now()
-		agg.Wait(make(chan bool))
-		waitTime := time.Now().Sub(start)
+			var r io.Reader
+			if tc.simulateErr {
+				r = iotest.TimeoutReader(testDataReader)
+			} else {
+				r = strings.NewReader("foo")
+			}
 
-		// Add buffer to avoid raciness due to processing time.
-		diffTime := waitTime - tc.expectExtraWait
-		if diffTime > testBufferDuration || diffTime < -1*testBufferDuration {
-			t.Errorf("Expected Wait() to wait the duration (%v) due to failed result, instead waited only %v", agg.retryWindow, waitTime)
-		}
+			err = agg.processResult(&plugin.Result{Body: r, NodeName: "node1", ResultType: "systemd_logs"})
+			if err == nil && tc.simulateErr {
+				t.Fatal("Expected error processing this due to reading error, instead got nil.")
+			}
+			// check time before/after wait and ensure it is greater than the retryWindow.
+			time.Sleep(tc.postProcessSleep)
+			start := time.Now()
+			agg.Wait(make(chan bool))
+			waitTime := time.Now().Sub(start)
+
+			// Add buffer to avoid raciness due to processing time.
+			diffTime := waitTime - tc.expectExtraWait
+			if diffTime > testBufferDuration || diffTime < -1*testBufferDuration {
+				t.Errorf("Expected Wait() to wait the duration %v (+/- %v), instead waited %v", tc.expectExtraWait, testBufferDuration, waitTime)
+			}
+		})
 	}
 }
 
 func TestAggregation_errors(t *testing.T) {
 	expected := []plugin.ExpectedResult{
-		{ResultType: "e2e"},
+		{ResultType: "e2e", NodeName: "global"},
 	}
 
 	withAggregator(t, expected, func(agg *Aggregator, srv *authtest.Server) {
 		resultsCh := make(chan *plugin.Result)
 		go agg.IngestResults(context.TODO(), resultsCh)
 
 		// Send an error
-		resultsCh <- pluginutils.MakeErrorResult("e2e", map[string]interface{}{"error": "foo"}, "")
+		resultsCh <- pluginutils.MakeErrorResult("e2e", map[string]interface{}{"error": "foo"}, "global")
 		agg.Wait(make(chan bool))
 
-		if result, ok := agg.Results["e2e"]; ok {
-			bytes, err := ioutil.ReadFile(path.Join(agg.OutputDir, result.Path()))
+		if result, ok := agg.Results["e2e/global"]; ok {
+			bytes, err := ioutil.ReadFile(path.Join(agg.OutputDir, result.Path(), "error.json"))
```

Comment:
Again, the error, being a single file, used to be named the same as the expected path. This would be counterintuitive to users who are not expecting users or even single files: when running e2e for instance you assume the directory will exist and you can `cd` to it and print files. In a CI script, that would fail if you didn't have the directory you expected.